### PR TITLE
Claude/review schema report m mq5 a

### PR DESCRIPTION
**Pull request by mohavro**
### **User description**

---
## Summary by cubic
Polished API/MCP docs and workflows, expanded auth/model tests, and applied repo-wide formatting. Startup errors are clearer, JWT now defaults to 30 minutes, the 3D view stays stable, a local Claude Bash allowlist is added, and 2D layout options were removed.

- **Refactors**
  - Simplified FastAPI lifespan and MCP server docstrings; made DATABASE_URL errors single-line; kept action version checks consistent while ignoring actions/checkout major drift.
  - Removed 2D Layout Type UI and related code; addressed a "variable defined multiple times" finding; added a TODO to improve empirical relationship formatting in the schema report.
  - Applied Autopep8/Black/isort/Prettier/Ruff formatting; added a local Claude Bash allowlist.

- **Tests**
  - Added unit suites for auth and Pydantic API models covering token expiry, hashing, repository flows, and validations.
  - Tightened integration checks: PR agent config/workflow assertions, Snyk triggers and SARIF upload, and branch/version consistency; stabilized fixtures and messages.
  - Verified uniform node color "#4ECDC4" in the enhanced 3D visualization.

<sup>Written for commit 568bef4e330b10050f59e51f2814667eabc07129. Summary will update on new commits. <a href="https://cubic.dev/pr/DashFin-FarDb/financial-asset-relationship-db/pull/936">Review in cubic</a></sup>

___

### **Description**
- Enhanced schema report generation logic and updated workflow configurations.
- Added comprehensive unit tests for authentication, covering JWT token generation and password hashing.
- Included unit tests for Pydantic models, ensuring validation and serialization.
- Improved configuration validation tests to ensure proper structure and types.
- Updated project status and structure details in documentation.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>tmp_merge_info.txt</strong><dd><code>Update workflow and schema report generation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tmp_merge_info.txt
<li>Updated workflow configuration for PR agent.<br> <li> Enhanced schema report generation logic.<br> <li> Fixed issues in frontend package configuration.<br>

</details>

  </td>
  <td><a href="https://github.com/DashFin-FarDb/financial-asset-relationship-db/pull/936/files#diff-da9db5afb91f12c028e8e1e330b2193199c70ba2381e3968b22b1d66016bd6bf">+570/-0</a>&nbsp; </td>
</tr>
</table></td></tr><tr><td><strong>Tests
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>test_api_auth.py</strong><dd><code>Comprehensive unit tests for authentication</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tests/unit/test_api_auth.py
<li>Added comprehensive unit tests for authentication.<br> <li> Covered JWT token generation and password hashing.<br> <li> Included tests for user repository and environment-based user seeding.<br> <br>

</details>

  </td>
  <td><a href="https://github.com/DashFin-FarDb/financial-asset-relationship-db/pull/936/files#diff-83902861e5041fe1844a00720a076666823ee5d7e147a2ed37c113d4e44cea5c">+737/-0</a>&nbsp; </td>
</tr>

<tr>
  <td>
    <details>
      <summary><strong>test_api_models.py</strong><dd><code>Unit tests for API models</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tests/unit/test_api_models.py
<li>Added unit tests for Pydantic models.<br> <li> Covered validation, serialization, and edge cases.<br>

</details>

  </td>
  <td><a href="https://github.com/DashFin-FarDb/financial-asset-relationship-db/pull/936/files#diff-3ddbe515282ee1e28033099d7d339659397eff551b72ad2b368103d711a3149b">+599/-0</a>&nbsp; </td>
</tr>

<tr>
  <td>
    <details>
      <summary><strong>test_pr_agent_config.py</strong><dd><code>Enhance configuration validation tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tests/integration/test_pr_agent_config.py
<li>Improved configuration validation tests.<br> <li> Ensured proper structure and types in configuration.<br>

</details>

  </td>
  <td><a href="https://github.com/DashFin-FarDb/financial-asset-relationship-db/pull/936/files#diff-037980cc9c030bfc186179cc11340ac89323de41ef09d59ba56f34135e076865">+30/-87</a>&nbsp; </td>
</tr>
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>systemManifest.md</strong><dd><code>Update project status and structure</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.elastic-copilot/memory/systemManifest.md
<li>Updated project status and structure details.<br> <li> Reflected changes in the number of files and project phases.<br>

</details>

  </td>
  <td><a href="https://github.com/DashFin-FarDb/financial-asset-relationship-db/pull/936/files#diff-75eef728bf319af894650e2e8e61a7ff92a977412aaccfaaaf8c94b0610d6ffb">+4/-3</a>&nbsp; &nbsp; &nbsp; </td>
</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **Penify usage**:
>Comment `/help` on the PR to get a list of all available Penify tools and their descriptions

___

## **CodeAnt-AI Description**
Add comprehensive auth and API model tests, tighten user-facing messages, and extend JWT expiry to 30 minutes

### What Changed
- Added extensive unit and integration tests covering authentication flows, JWT handling, password hashing, user repository behavior, and Pydantic API models; these tests validate valid/expired/invalid tokens, environment seeding, model validation, and many edge cases.
- Increased default JWT access token lifetime from 15 minutes to 30 minutes (tokens now include an exp claim reflecting the longer default).
- Made startup and configuration error messages clearer: the DATABASE_URL missing error and the MCP server missing-dependency CLI message now present single-line, direct text for easier reading in logs and terminals.
- Cleaned and standardized test assertions and documentation-related test outputs so failures report concise, single-line messages (improves readability of test failures and CI logs).

### Impact
`✅ More reliable test coverage`
`✅ Clearer startup and configuration errors`
`✅ Longer default session tokens (30 min)`
<details>
<summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

### CODE REVIEW

File: tests/unit/test_api_auth.py [hunk 1]
```diff
@@ -0,0 +1,740 @@
...
+
+        assert exc_info.value.status_code == 401
+        assert "Could not validate credentials" in exc_info.value.detail
+
+    @pytest.mark.asyncio
+    async def test_get_current_user_missing_sub_claim(self):
+        """get_current_user raises HTTPException when sub claim is missing."""
+        # Token without 'sub' claim
+        data = {"other": "value"}
+        token = create_access_token(data)
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/unit/test_api_auth.py]:
```suggestion
```

File: tests/unit/test_api_auth.py [hunk 2]
```diff
@@ -0,0 +1,739 @@
+"""
+Comprehensive unit tests for api/auth.py
+
+Tests cover authentication, JWT token generation, password hashing,
+user repository, and environment-based user seeding.
+"""
+
+from __future__ import annotations
+
+import os
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/unit/test_api_auth.py]:
```suggestion
```

File: tests/unit/test_api_auth.py [hunk 3]
```diff
@@ -0,0 +1,739 @@
...
+"""
+
+from __future__ import annotations
+
+from datetime import datetime, timedelta, timezone
+from unittest.mock import Mock, patch
+
+import pytest
+from fastapi import HTTPException
+from jwt import ExpiredSignatureError, InvalidTokenError
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/unit/test_api_auth.py]:
```suggestion
from jwt import InvalidTokenError
```

File: .elastic-copilot/memory/dependencyMatrix.md [hunk 4]
```diff
@@ -48,8 +43,20 @@ Top dependencies:
...
+- ../../app/types/api
+- ../test-utils
+- ../../app/components/MetricsDashboard
+- ../../app/components/AssetList
+
+### JS
+
+Top dependencies:
+
+- +i),a)}}catch(t){n(t.toString())}},Et.prototype.getAvailableImages=function(t){var e=this.availableImages[t];return e||(e=[]),e},Et.prototype.getLayerIndex=function(t){var e=this.layerIndexes[t];return e||(e=this.layerIndexes[t]=new n),e},Et.prototype.getWorkerSource=function(t,e,r){var n=this;if(this.workerSources[t]||(this.workerSources[t]={}),this.workerSources[t][e]||(this.workerSources[t][e]={}),!this.workerSources[t][e][r]){var i={send:function(e,r,i){n.actor.send(e,r,i,t)}};this.workerSources[t][e][r]=new this.workerSourceTypes[e](<i,this.getLayerIndex(t),this.getAvailableImages(t)>)}return this.workerSources[t][e][r]},Et.prototype.getDEMWorkerSource=function(t,e){return this.demWorkerSources[t]||(this.demWorkerSources[t]={}),this.demWorkerSources[t][e]||(this.demWorkerSources[t][e]=new u),this.demWorkerSources[t][e]},Et.prototype.enforceCacheSizeLimit=function(e,r){t.enforceCacheSizeLimit(r)},typeof WorkerGlobalScope<
```
[the remark this answers is not in the document]
Comment (mohavro) [reply .elastic-copilot/memory/dependencyMatrix.md]:
```suggestion
```
